### PR TITLE
Show account balance in signature screen

### LINKED ISSUE #5928: [Bug] Account Balance is not displayed on any Signature screen
**Describe the bug**
Account balance is not displayed on any of the Signature screens.

**Screenshots**

image

**To Reproduce**
_Steps to reproduce the behavior_
1. Go to 'test dapp https://metamask.github.io/test-dapp/
2. Connect MM
3. Click on any signature
4. See no balance is displayed on the header

**Expected behavior**
We expect to see the account balance, in the same way we do on Extension.

Screenshot from 2023-03-08 08-21-13

**Smartphone (please complete the following information):**

- Device: Pixel 6
- OS: Android 12
- App Version: v6.0.1 (1078)

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

### PR DESCRIPTION
**Pull request by segun**
**Description**

On signing page, show current balance.

**Screenshots/Recordings**

**Before**

**After**
Screenshot 2023-05-09 at 17 30 01

**Issue**
Fixes: #5928 

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Comment by bschorchit** [comment]
Thank you for pushing this improvement forward! 🔥
Could we remove these components (see screenshot) from the confirmation screen? They are redundant now that we have introduced the new header 😊

Screenshot 2023-05-04 at 18 15 11


Another ask, could we move the `Sign this message?`to below the header and above the message? That way it'll have consistency with the transaction confirmations screens where origin pill + header are always on the top.

Screenshot 2023-05-04 at 18 17 35

Thank you! ❤️

File: ios/Podfile.lock [hunk 1]
```diff
@@ -874,4 +874,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 74e503a275b4ed097cb01c171027882e9cdc699d
 
-COCOAPODS: 1.11.2
+COCOAPODS: 1.11.3
```
Comment (jpuri) [inline ios/Podfile.lock:877]: 🔪

**Review by jpuri — DISMISSED** [review]
Changes look good.

Please revert changes in [ios/Podfile.lock](https://github.com/MetaMask/metamask-mobile/pull/6192/files#diff-b7d85c35701918f25c9b2c418249f9b66fd9b067676e295a9eff493b73ee1752) before merging.

**Review by blackdevelopa — COMMENTED** [review]
left a couple tiny comments

**Review by blackdevelopa — DISMISSED** [review]

File: app/components/UI/AccountInfoCard/index.test.tsx [hunk 2]
```diff
@@ -28,21 +44,42 @@ const initialState = {
       },
       NetworkController: {
         providerConfig: {
+          chainId: '0x1',
+          type: 'ropsten',
+          nickname: 'Ropsten',
```
[the remark this answers is not in the document]
Comment (segun) [reply app/components/UI/AccountInfoCard/index.test.tsx]: Fixed [here](https://github.com/MetaMask/metamask-mobile/commit/e5985607c46cb10e8994f18f31ecaad0c451d6fc)

**Review by blackdevelopa — APPROVED** [review]

**Review by jpuri — APPROVED** [review]

**Review by jpuri — APPROVED** [review]

**Comment by seaona** [comment]
Hey! This PR has been merged before going over QA. I've found several issues, that need to be addressed

## Personal Sign is broken
Whenever I try to perform a personal Sign MetaMask does not open and I see the following error:

image

## Domain not displayed
The domain is not displayed on any of the signature types

image

**Comment by seaona** [comment]
I've opened 2 separate issues for solving that:
- https://github.com/MetaMask/metamask-mobile/issues/6428
- https://github.com/MetaMask/metamask-mobile/issues/6429

**Comment by bschorchit** [comment]
I've created https://github.com/MetaMask/metamask-mobile/issues/6430 to address this part as well 

> Another ask, could we move the Sign this message?to below the header and above the message? That way it'll have consistency with the transaction confirmations screens where origin pill + header are always on the top.

**Pull request merged**